File: Striver/Recursion/q13.java
package Striver.Recursion;

public class q13 {
    // print the count of subsequences which sums to k
    public static int printSubsequence(int[] arr, int k, int i, String ans) {
        if (i == arr.length) {
            if (k == 0) {
                System.out.println(ans);
                return 1;
            }
            return 0;
        }
        int count = 0;
        count += printSubsequence(arr, k - arr[i], i + 1, ans + arr[i] + " ");
        count += printSubsequence(arr, k, i + 1, ans);
        return count;
    }

    public static void main(String[] args) {
        int[] arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9 };
        int k = 10;
        System.out.println(printSubsequence(arr, k, 0, ""));
    }
}

File: Striver/Recursion/q13.py
# print the count of subsequences which sums to k

def countSubsequences(arr, k, index, sum):
    if index == len(arr):
        if sum == k:
            return 1
        return 0
    count = countSubsequences(arr, k, index + 1, sum)
    count += countSubsequences(arr, k, index + 1, sum + arr[index])
    return count

arr = [1, 2, 3, 4, 5]
print(countSubsequences(arr, 5, 0, 0))